### ISSUE TITLE
TransactionScope for EFUtilities

### ISSUE THREAD

**Issue by Manish176837** [MikaelEliasson/EntityFramework.Utilities#43]
I am using EF.Utilities 0.5.0 in my code and have some query regarding it.
First of all, this is the best package that I have come across for performing bulk operations.
Now about my query, I have a condition in my code where I am trying to delete an entire table (around 24K records) and then add same number of records again.
For deletion, I have added following code:
EFBatchOperation.For(context, context.ArticleImages).Where(x => true).Delete();
And for insertion, I am using following code:
EFBatchOperation.For(context, context.ArticleImages).InsertAll(articleImages);

I am doing this in a single method and passing same context object in both calls.
Now, I want to include these both operations in a single TransactioScope. Please suggest how should I do it?
Thanks in advance.

**Comment by MikaelEliasson** [comment]
First of all I would suggest you to actually test that the following works by throwing an exception between the calls and make sure nothing was deleted. I find Transaction support slightly hard to test.

Basically you should only need to wrap the whole block in a TransactionScope. See the example here: https://github.com/MikaelEliasson/EntityFramework.Utilities/issues/26

**Comment by Manish176837** [comment]
Hi Mikael,
Thanks for the instant reply.
I actually tried doing that but got exception saying "User transactions not supported by EF".
This is the link that was given in the message of the exception: https://msdn.microsoft.com/en-us/data/dn307226.

**Comment by Manish176837** [comment]
Hi Mikael,

I am able to set the transaction scope for my code now.
Actually, the solution was given in the same link that I shared above. All I had to do is suspend the retrying execution strategy for my code. I followed the steps and added the same flow(created new TransactionScope object --> Created context --> used EFUtilities functionality).
This is now working perfectly.
Thanks for your help :)
